builtin/logical/kv/path_metadata.go

```diff
@@ -486,14 +511,33 @@ func (b *versionedKVBackend) pathMetadataWrite() framework.OperationFunc {
 		if err != nil {
 			return nil, err
 		}
+
+		now := ptypes.TimestampNow()
 		if meta == nil {
-			now := ptypes.TimestampNow()
 			meta = &KeyMetadata{
-				Key:         key,
-				Versions:    map[uint64]*VersionMetadata{},
-				CreatedTime: now,
-				UpdatedTime: now,
+				Key:                    key,
+				Versions:               map[uint64]*VersionMetadata{},
+				CreatedTime:            now,
+				UpdatedTime:            now,
+				CurrentMetadataVersion: 0, // Initialize to 0 for new metadata
```

Comment:
```suggestion
				CurrentMetadataVersion: 1, // Initialize to 1 for new metadata
```

Do we want to go to 1? It allows us to differentiate new versus upgraded entries in case there is some bug. From 1+, everything should behave the same.

Reply:
In particular, based on the comment on the lines above (below in the GitHub viewer), you could set this to zero, move the check below about CAS value to global, and _then_ increment, so zero means entry does not exist and/or is upgraded from earlier, and then always increment (getting 1 on net-new entries). Caller would then provider `0` as the CAS value for creation, and get an entry with 1, which is expected.